viewer-prototype/src/browser/trace-viewer/components/table-output-component.tsx

```diff
@@ -2,78 +2,177 @@
 import { AbstractOutputComponent, AbstractOutputProps, AbstractOutputState } from './abstract-output-component';
 import * as React from 'react';
 import { AgGridReact } from 'ag-grid-react';
-import { ColDef } from 'ag-grid-community';
+import { ColDef, IDatasource, GridReadyEvent } from 'ag-grid-community';
 import { Entry, EntryHeader } from 'tsp-typescript-client/lib/models/entry';
 import { QueryHelper } from 'tsp-typescript-client/lib/models/query/query-helper';
+import { cloneDeep } from 'lodash';
 
 type TableOuputState = AbstractOutputState & {
     tableColumns: ColDef[];
-    tableLines: any[];
 };
 
-export class TableOutputComponent extends AbstractOutputComponent<AbstractOutputProps, TableOuputState> {
+type TableOutputProps = AbstractOutputProps & {
+    cacheBlockSize?: number;
+    maxBlocksInCache?: number;
+    columnWidth?: number;
+    blockLoadDebounce?: number;
+};
+
+export class TableOutputComponent extends AbstractOutputComponent<TableOutputProps, TableOuputState> {
+    private cacheBlockSize: number;
+    private maxBlocksInCache: number;
+    private blockLoadDebounce: number;
+    private debugMode = false;
+    private columnIds: Array<number> = [];
+    private fetchColumns = true;
+    private columnArray = new Array<any>();
+    private showIndexColumn = false;
+    private columnWidth: number;
+    private components: any;
 
-    constructor(props: AbstractOutputProps) {
+    constructor(props: TableOutputProps) {
         super(props);
+
+        this.cacheBlockSize = this.props.cacheBlockSize ? this.props.cacheBlockSize : 200;
+        this.maxBlocksInCache = this.props.maxBlocksInCache ? this.props.maxBlocksInCache : 5;
+        this.blockLoadDebounce = this.props.blockLoadDebounce ? this.props.blockLoadDebounce : 250;
+        this.columnWidth = this.props.columnWidth ? this.props.columnWidth : 200;
```

Comment:
instead of assigning local fields, you could add defaultProps values using " static defaultProps: Partial<FilterTreeProps> = { columnWidth = 200}", etc and use this.props.columnWidth instead

Reply:
Done.